app/models/manageiq/providers/microsoft/infra_manager.rb

```diff
@@ -101,9 +101,7 @@ def vm_resume(vm, _options = {})
   end
 
   def vm_destroy(vm, _options = {})
-    if vm.power_state == "on"
-      vm_stop(vm)
-    end
+    vm_stop(vm)
```

Comment:
@bronaghs Is this vm_stop call synchronous?  If not are we still at risk of hitting the issue of the VM running when we try to remove it?

Also, is calling stop on an already stopped VM handled properly?  (i.e. does not raise an error)